api4/channel_test.go

```diff
@@ -1247,37 +1248,38 @@ func TestDeleteChannel(t *testing.T) {
 	user2 := th.BasicUser2
 
 	// successful delete of public channel
-	publicChannel1 := th.CreatePublicChannel()
-	pass, resp := Client.DeleteChannel(publicChannel1.Id)
-	CheckNoError(t, resp)
-
-	require.True(t, pass, "should have passed")
+	th.TestForAllClients(t, func(t *testing.T, client *model.Client4) {
+		publicChannel1 := th.CreatePublicChannel()
+		pass, resp := client.DeleteChannel(publicChannel1.Id)
+		CheckNoError(t, resp)
 
-	ch, err := th.App.GetChannel(publicChannel1.Id)
-	require.True(t, err != nil || ch.DeleteAt != 0, "should have failed to get deleted channel, or returned one with a populated DeleteAt.")
+		require.True(t, pass, "should have passed")
 
-	post1 := &model.Post{ChannelId: publicChannel1.Id, Message: "a" + GenerateTestId() + "a"}
-	_, resp = Client.CreatePost(post1)
-	require.NotNil(t, resp, "expected response to not be nil")
+		ch, err := th.App.GetChannel(publicChannel1.Id)
+		require.True(t, err != nil || ch.DeleteAt != 0, "should have failed to get deleted channel, or returned one with a populated DeleteAt.")
 
-	// successful delete of private channel
-	privateChannel2 := th.CreatePrivateChannel()
-	_, resp = Client.DeleteChannel(privateChannel2.Id)
-	CheckNoError(t, resp)
+		post1 := &model.Post{ChannelId: publicChannel1.Id, Message: "a" + GenerateTestId() + "a"}
+		_, resp = th.Client.CreatePost(post1)
```

Comment:
This should be the client injected by the helper:

```suggestion
		_, resp = client.CreatePost(post1)
```